client/src/app/core/poll.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable } from '@angular/core';
+
+/**
+ * The possible keys of a poll object that represent numbers.
+ * TODO Should be 'key of MotionPoll if type of key is number'
+ * TODO: normalize MotionPoll model and other poll models
+ * TODO: reuse more motion-poll-service stuff
+ */
+export type CalculablePollKey = 'votesvalid' | 'votesinvalid' | 'votescast' | 'yes' | 'no' | 'abstain';
+
+/**
+ * Shared service class for polls.
+ * TODO: For now, motionPolls only. TODO See if reusable for assignment polls etc.
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class PollService {
+    /**
+     * The chosen and currently used base for percentage calculations. Is set by
+     * the config service
+     */
+    public percentBase: string;
+
+    /**
+     * The default majority method (as set per config).
+     */
+    public defaultMajorityMethod: string;
+
+    /**
+     * An array of value - label pairs for special value signifiers.
+     * TODO: Should be given by the server, and editable. For now: hard coded
+     */
+    private _specialPollVotes: [number, string][] = [[-1, 'Majority'], [-2, 'Undocumented']];
```

Comment:
the config varbiable should already exist.
For motion poll it would be:
`motions_poll_default_majority_method`

Reply:
That is something else (the choice if the default method used for the quorum calculations). I need the 'special values that are not really numbers'.

Reply:
OK